Lib/configparser.py

```diff
@@ -1037,6 +1059,13 @@ def _read(self, fp, fpname):
                 cursect[optname].append(value)
             # a section header or option header?
             else:
+                if self._allow_unnamed_section and cursect is None:
+                    sectname = UNNAMED_SECTION
+                    cursect = self._dict()
+                    self._sections[sectname] = cursect
+                    self._proxies[sectname] = SectionProxy(self, sectname)
+                    elements_added.add(sectname)
+
```

Comment:
I think this whole block should be moved down after the "is it a section header?" check, i.e. before the "no section header?" case. Otherwise, it seems that an (empty) UNNAMED_SECTION will be added to the config even if the first line *is* a section header.

Reply:
I tried but then it handles the first line as a section instead of an option.